query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java

```diff
@@ -131,5 +178,42 @@ static String resolveStatement(String query, Params params) {
     private static String getStringParam(String value) {
       return "'" + value.replace("'", "''") + "'";
     }
+
+    @SneakyThrows
+    Observable<Row> convert(ResultSet resultSet) {
+      List<Row.Builder> rowBuilderList = new ArrayList<>();
+      while (resultSet.next()) {
```

Comment:
Seems like this would be cleaner if streamed and immutable.

Reply:
result set is not streamable